### DISCUSSION TITLE
Increasing Turbulence on Large scales

### DISCUSSION THREAD

**Discussion by rakyuv** [firemodels/fds#13868] [General]
**Describe the bug**
I am interested to create the atmospheric turbulent flow with FDS, with a domain of 90x90x40 m^3. I have a horizontal resolution of 20 cm and a vertical resolution of 25 cm. I ran the simulation for 2400 s, which took approximately 12 hours to simulate. I plot the energy spectrum at 2.5 m from the surface from FDS, and when it is compared with a real instrument that measured the 3d wind, I find that the FDS results weren't so energetic. This does mean that the flow in FDS is not turbulent enough. 

I had tried increasing the domain size by degrading the resolution away from the source. I ran this for 2400 s, but the spectrum didn't show any improvement. In fact the simulation with large domain had less energy in the TKE spectrum compared to the smaller domain with constant 20 cm resolution. 

Presently, I'm tryintg to run for 5000, 7500 and 10000 seconds to see if running for a longer time with PERIODIC FLOW ONLY in the lateral boundaries would make the flow more turbulent and therefore have a spectrum closer to that of the atmosphere at that altitude. I don't have the results yet. 

If the long time-series doesn't result in a turbulent flow, then I have no idea on how to increase the turbulence in the flow. 

**Desktop:**
 - OS: Linux
 - Version: HPC

**Additional context**
Could you please tell me how to increase the turbulence in the flow field ? I'm yet to try synthetic turbulence though

**Comment by rmcdermo** [comment]
You are in research territory.  FDS does not have many atmospheric validation cases, but the [Schoenberg_Ekman_Layers](https://github.com/firemodels/fds/tree/master/Validation/Schoenberg_Ekman_Layers/FDS_Input_Files) would not be a bad place to start.

With 20 cm resolution, you have plenty of resolution to get turbulent flow.  You have not shown us anything related to the flow field, a slice of velocity, a plot of the velocity components in time, for example.  My bet is that you have not read the user manual to know that DEVC quantities are by default time averaged.  And that the default interval is (T_END-T_BEGIN)/1000.  This would give you a smoothing of about 2.4 s, which may dampen any TKE you are seeing.

Obviously, you need to know and apply the correct boundary conditions.  You also need to understand how to turn a velocity signal into a TKE spectrum, which is not trivial, so there could be errors there.

I am transferring this to a discussion, as you have not demonstrated any "issue" with the code.

**Comment by rakyuv** [comment]
Thank you @rmcdermo for your answer. My apologies for posting it as a bug, and for not giving enough information to explain the problem. 

Here is the spectrum I talked about:
![spectrum](https://github.com/user-attachments/assets/16ead784-0cfc-432e-9125-01f80488fd25)

For the results, I output the values using SLCF 3D, with DT_SL3D=1. In the above figure, I plotted the instrument readings at (20 Hz) for a 5-minute window alongside FDS TKE at 1Hz, for each multiple 5 minutes window.

I also plotted the -5/3 line on top of each spectrum

**Comment by rakyuv** [comment]
Hi,

I have found the fault in my comparison. As you said @rmcdermo, the 20 cm resolution is good enough for turbulent flow simulations. However, my sampling frequency in FDS of 1 Hz was hardly any match to the sampling frequency of the instrument which measured the wind at 20 Hz. 

Therefore, I undersampled the wind measurement datasets to 1 Hz and compared with FDS, where the difference between the two spectras were quite small. I am also trying to increase the sampling frequency of FDS to see if it changes anything. 

For the remaining part, I have added Jarrin's synthetic Turbulence, and would give the integral length scales L_ij for L_EDDY, RMS of velocity for V_RMS, and N_eddy to see if I can get the right amount of turbulence. 

Thanks for your help.
